### PR TITLE
Fix omitempty tags for optional API fields

### PR DESCRIPTION
**Pull request by nikhita**
Fixes omitempty json tags for optional API fields

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/authorization/v1/types.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ type SelfSubjectRulesReview struct {
 
 type SelfSubjectRulesReviewSpec struct {
 	// Namespace to evaluate rules for. Required.
-	Namespace string `json:"namespace,omitempty" protobuf:"bytes,1,opt,name=namespace"`
+	Namespace string `json:"namespace" protobuf:"bytes,1,opt,name=namespace"`
```
Comment (nikhita) [inline staging/src/k8s.io/api/authorization/v1/types.go]: This is a breaking change...I'm not sure if that's ok.

Comment (liggitt) [reply staging/src/k8s.io/api/authorization/v1/types.go]: What happens today if you submit this object without a namespace field?

Comment (liggitt) [reply staging/src/k8s.io/api/authorization/v1/types.go]: > What am I doing wrong?

nothing. it is not a persisted object, and is not given a name. adding `--v=8` would show the response containing the rules you asked for.

Comment (liggitt) [reply staging/src/k8s.io/api/authorization/v1/types.go]: > **Without namespace**:
> 
> ```yaml
> apiVersion: authorization.k8s.io/v1
> kind: SelfSubjectRulesReview
> metadata:
>   name: foo
> spec:
> ```
> 
> ```
> error: error validating "foo.yaml": error validating data: ValidationError(SelfSubjectRulesReview): missing required field "spec" in io.k8s.api.authorization.v1.SelfSubjectRulesReview; if you choose to ignore these errors, turn validation off with --validate=false
> ```
> 
> So should be safe. :+1: 

that's client-side validation. what does the server do if you submit that object anyway (`--validate=false`)

Comment (nikhita) [reply staging/src/k8s.io/api/authorization/v1/types.go]: > that's client-side validation. 

oh yes. :woman_facepalming: 

> what does the server do if you submit that object anyway

`Error from server (BadRequest): error when creating "foo.yaml": no namespace on request`

phew. :)

**Pull request closed without merging**